ci/scripts/check-bazel-tags.sh

```diff
@@ -6,14 +6,125 @@
+      //...
+  )" \
+  --output=label_kind)
+check_empty "${untagged}" \
+"Target(s) above depend(s) on //hw:verilator, please tag it with verilator or
+(to prevent matching any wildcards) manual.
+NOTE: test_suites that contain targets with different tags should almost
+universally use the manual tag."
+
+# This check ensures Opentitan software can be built with wildcards in
+# environments that don't have vivado or vivado tools installed by using
+# --build_tag_filters=-vivado.
+untagged=$(./bazelisk.sh query \
+  "rdeps(
+      //...,
+      kind(
+          'bitstream_splice',
+          //...
+      )
+  )
+  except
+  attr(
+      tags,
+      '$(exact_regex "(cw310_rom|cw310_test_rom|vivado|manual)")',
+      //...
+  )" \
+  --output=label_kind)
+check_empty "${untagged}" \
+"Target(s) above depend(s) on a bitstream_splice that isn't cached.
+Please tag it with vivado or (to prevent matching any wildcards) manual.
+NOTE: test_suites that contain targets with different tags should almost
+universally use the manual tag."
+
+# This check ensures cw310 users may group tests that depend on the cached
+# cw310_test_rom bitstream.
+mistagged=$(./bazelisk.sh query \
+    "tests(`# Only output tests`
+        rdeps(
+            attr(`# Anything tagged cw310_test_rom`
+                tags,
+                '$(exact_regex cw310_test_rom)',
+                tests(//...)
+            ),
+            kind(`# That depends on a bitstream splice`
+                'bitstream_splice',
+                //...
+            )
+            except`# Other than those used to build the test_ROM bitstream`
+            deps(
+                //hw/bitstream:test_rom
+            )
+        )
+    )" \
+    --output=label_kind)
+check_empty "${mistagged}" \
+"Target(s) above depend(s) on a bitstream_splice rule other than those used to
+generate the cached bitstream with the test_ROM, but is tagged with cw310_test_rom.
+Please either:
+-Correct the dependencies to exclude other bitstream_splices,
+-Correct the target by setting it to something other than cw310_test_rom,
+-Remove the cw310_test_rom tag.
+NOTE: test_suites that contain targets with different tags should almost
+universally use the manual tag."
+
+
+
+# This check ensures cw310 users may group tests that depend on the cached
+# cw310_rom bitstream.
+mistagged=$(./bazelisk.sh query \
+    "tests(`# Only output tests`
+        rdeps(
+            attr(`# Anything tagged cw310_rom`
+                tags,
+                '$(exact_regex cw310_rom)',
+                //...
+            ),
+            kind(`# That depends on a bitstream splice`
+                'bitstream_splice',
+                //...
+            )
+            except`# Other than those used to build the ROM bitstream`
+            deps(//hw/bitstream:rom)
+        )
+    )" \
+    --output=label_kind)
+check_empty "${mistagged}" \
+"Target(s) above depend(s) on a bitstream_splice rule other than those used to
```

Comment:
Nit: s/rule/target

Reply:
Thanks, target is correct but also overloaded in our context.
There's a var named target that's used in opentitan_functest, and it's used in the system testing context and hardware description context along with device.
I'm going to use "bazel target" here to try to keep things clearer.

Reply:
and functest target for the other case in use here

Reply:
I'll keep the "Target above" references as is because the context makes it very clear.